kitsune/customercare/zendesk.py

```diff
@@ -42,6 +50,21 @@ def _user_to_zendesk_user(self, user, email):
             external_id=external_id,
         )
 
+    def get_user_by_email(self, email):
+        """Given an email, return a user from Zendesk."""
+        # This returns a generator, but we only want/expect one user
+        # If it returns more than one, we should fail
+        # Otherwise return the Zendesk user object
+        search_results = self.client.search(type="user", query=f"email:{email}")
+
+        user_found = None
+        for user in search_results:
+            if user_found is not None:
+                raise ValueError(f"Found more than one user with email {email}")
```

Comment:
The search feature only ever returns a generator. One result or 100, you get a generator. Since we are looking up an e-mail and that is a primary identifier, we should never expect more than one response - although the API would allow it! With no other way to return a search result, we have to live with the generator object.